x-pack/metricbeat/module/azure/compute_vm/client_helper.go

```diff
@@ -32,8 +35,14 @@ func mapMetric(client *azure.Client, metric azure.MetricConfig, resource resourc
 		if len(*metricDefinitions.Value) == 0 {
 			return nil, errors.Errorf("no metric definitions were found for resource %s and namespace %s.", *resource.ID, *namespace.Properties.MetricNamespaceName)
 		}
+		var filteredMetricDefinitions []insights.MetricDefinition
+		for _, metricDefinition := range *metricDefinitions.Value {
+			if (metricDefinition.Name.LocalizedValue != nil && !strings.Contains(*metricDefinition.Name.LocalizedValue, "(Deprecated)")) || metricDefinition.Name.LocalizedValue == nil {
```

Comment:
I think this would be simpler and equivalent
```suggestion
			if metricDefinition.Name.LocalizedValue == nil || !strings.Contains(*metricDefinition.Name.LocalizedValue, "(Deprecated)")) {
```

Btw, why are we filtering out deprecated values?

Can the use of "localized" values be a problem if languages different to English are used?

Reply:
It seems that the API will not return the "deprecated" message although it will do so for the "preview". Non the less, testing different languages for the metric localized name, indeed should translate the "deprecated" word as well (https://docs.microsoft.com/nl-NL/azure/azure-monitor/platform/metrics-supported#microsoftcomputevirtualmachines , for nl version "afgeschaft"). I have remove the entire condition in both places.